### PR TITLE
[pytorch][sagemaker] Prep for PT 1.13 SM release

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Preparing for release of PyTorch 1.13 SageMaker Inference image

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [x] Revision B: `sagemaker_remote_tests = "rc"`
- [x] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "torch": [
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json:4]: Also, you say that `"The binaries are patched for this CVE"`. Has the patch been made by upstream?

File: pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "cvss_v2_score": 0.0, 
+        "cvss_v3_severity": "CRITICAL", 
+        "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html", 
+        "source": "UBUNTU_CVE", 
+        "severity": "MEDIUM", 
+        "status": "ACTIVE", 
+        "title": "CVE-2022-45907 - torch"
+    }, 
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json:29]: Same as above, can we rename this to `reason_to_ignore`

File: pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{
+  "torch": [
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json:4]: Same

File: pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        "cvss_v2_score": 0.0, 
+        "cvss_v3_severity": "CRITICAL", 
+        "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html", 
+        "source": "UBUNTU_CVE", 
+        "severity": "MEDIUM", 
+        "status": "ACTIVE", 
+        "title": "CVE-2022-45907 - torch"
+    }, 
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json:29]: Same

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{
+  "torch": [
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json:4]: Same

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      "cvss_v2_score": 0.0,
+      "cvss_v3_severity": "CRITICAL",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html",
+      "source": "UBUNTU_CVE",
+      "severity": "MEDIUM",
+      "status": "ACTIVE",
+      "title": "CVE-2022-45907 - torch"
+    },
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json:29]: Same

File: pytorch/inference/docker/1.13/py3/Dockerfile.cpu [hunk 7]
```diff
@@ -4,7 +4,7 @@ ARG MAMBA_VERSION=22.9.0-1
 ARG OPEN_MPI_VERSION=4.1.4
 
 # PyTorch Binaries
-ARG TORCH_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.13.0_ec2/20221105-013447/41160d617a294a8963a868ec942a2569a2fbf3ea/torch-1.13.0%2Bcpu-cp39-cp39-linux_x86_64.whl
+ARG TORCH_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.13.0_ec2/20221205-202607/8cf843550346728d7e661cd0a26f74b603dc17a5/torch-1.13.0%2Bcpu-cp39-cp39-linux_x86_64.whl
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/Dockerfile.cpu:7]: The binary name says `r1.13.0_ec2`. Does it just correspond to `ec2` images or `SM` as well?

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 8]
```diff
@@ -8,7 +8,7 @@ ARG CUBLAS_VERSION=11.10.3.66
 ARG CUDNN_VERSION=8.5.0.96
 
 # PyTorch Binaries and versions.
-ARG TORCH_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.13.0_ec2/20221105-023216/41160d617a294a8963a868ec942a2569a2fbf3ea/torch-1.13.0%2Bcu117-cp39-cp39-linux_x86_64.whl
+ARG TORCH_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r1.13.0_ec2/20221205-210214/8cf843550346728d7e661cd0a26f74b603dc17a5/torch-1.13.0%2Bcu117-cp39-cp39-linux_x86_64.whl
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:11]: The binary name says `r1.13.0_ec2`. Does it just correspond to `ec2` images or `SM` as well?

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json [hunk 9]
```diff
@@ -0,0 +1,54 @@
+{
+  "torch": [
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json:4]: Same

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+      "cvss_v2_score": 0.0,
+      "cvss_v3_severity": "CRITICAL",
+      "source_url": "https://people.canonical.com/~ubuntu-security/cve/2022/CVE-2022-45907.html",
+      "source": "UBUNTU_CVE",
+      "severity": "MEDIUM",
+      "status": "ACTIVE",
+      "title": "CVE-2022-45907 - torch"
+    },
+    {
+      "//Reason_for_Allow": "The binaries are patched for this CVE, but the system checks version and not code.",
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json:29]: Same

File: test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py [hunk 11]
```diff
@@ -119,7 +119,7 @@ def test_pytorch_inference_torchdata_gpu(
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_inference)
     if test_utils.is_image_incompatible_with_instance_type(pytorch_inference, ec2_instance_type):
         pytest.skip(f"Image {pytorch_inference} is incompatible with instance type {ec2_instance_type}")
-    if Version(image_framework_version) in SpecifierSet(">=1.11,<1.13"):
+    if Version(image_framework_version) in SpecifierSet(">=1.11,<=1.13"):
```
Comment (shantanutrip) [inline test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py:122]: Is there a reason for the PT1.13 tests using Dev repo?

File: pytorch/buildspec.yml [hunk 12]
```diff
@@ -46,77 +46,77 @@ context:
...
+  #   device_type: &DEVICE_TYPE gpu
+  #   python_version: &DOCKER_PYTHON_VERSION py3
+  #   tag_python_version: &TAG_PYTHON_VERSION py39
+  #   cuda_version: &CUDA_VERSION cu117
+  #   os_version: &OS_VERSION ubuntu20.04
+  #   tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION,
+  #                "-ec2-example" ]
+  #   docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example, /Dockerfile., *DEVICE_TYPE ]
+  #   context:
+  #     <<: *TRAINING_CONTEXT
```
Comment (shantanutrip) [inline pytorch/buildspec.yml:119]: The PR comment says  that we are planning to go with `Sagemaker` release. Is there a reason that `ec2` images have not been commented on the buildspec?

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 8 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:11]: Inference  images only use one type of binary, being EC2, for both EC2 and SageMaker. Training images need a separate  binary for SageMaker.

File: test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py (hunk 11 above)
Comment (xncqr) [reply test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py:122]: it's only labeled that way and temporary till the upstream solves the tagging items issue in the S3 buckets. This is already used by other training images.

File: pytorch/buildspec.yml (hunk 12 above)
Comment (xncqr) [reply pytorch/buildspec.yml:119]: it says prep, but need the ec2 included to handle patching as well.

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json (hunk 10 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json (hunk 9 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.sagemaker.gpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json (hunk 2 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json (hunk 3 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json (hunk 4 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/Dockerfile.sagemaker.cpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json (hunk 5 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.ec2.gpu.os_scan_allowlist.json]: updated

File: pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/Dockerfile.ec2.cpu.os_scan_allowlist.json]: We patched it from https://github.com/pytorch/pytorch/commit/767f6aa49fe20a2766b9843d01e3b7f7793df6a3 due to the severity.. will likely be included in 1.13.1 which is possibly slated for the 17th of December, but not hard set

**Review by shantanutrip — DISMISSED** [review]
LGTM 🚢
